bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveSwitchAllCommandClass.java

```diff
@@ -0,0 +1,207 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.zwave.internal.protocol.commandclass;
+
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+import com.thoughtworks.xstream.annotations.XStreamOmitField;
+import org.openhab.binding.zwave.internal.config.ZWaveDbCommandClass;
+import org.openhab.binding.zwave.internal.protocol.SerialMessage;
+import org.openhab.binding.zwave.internal.protocol.ZWaveController;
+import org.openhab.binding.zwave.internal.protocol.ZWaveEndpoint;
+import org.openhab.binding.zwave.internal.protocol.ZWaveNode;
+import org.openhab.binding.zwave.internal.protocol.commandclass.ZWaveCommandClass;
+import org.openhab.binding.zwave.internal.protocol.event.ZWaveCommandClassValueEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handles the Switch All command class. Sends all on or all off commands to device
+ * 
+ * @author Pedro Paixao
+ * @since 1.8.0
+ */
+@XStreamAlias(value="switchAllCommandClass")
+public class ZWaveSwitchAllCommandClass extends ZWaveCommandClass {
+    @XStreamOmitField
+    private static final Logger logger = LoggerFactory.getLogger(ZWaveSwitchAllCommandClass.class);
+   
+    private static final int SWITCH_ALL_SET = 0x01;
+    private static final int SWITCH_ALL_GET = 0x02;
+    private static final int SWITCH_ALL_REPORT = 0x03;
+    private static final int SWITCH_ALL_ON = 0x04;
+    private static final int SWITCH_ALL_OFF = 0x05;
+    
+    private static final int SWITCH_ALL_EXCLUDED = 0x00;
+    private static final int SWITCH_ALL_INCLUDE_ON_ONLY = 0x01;
+    private static final int SWITCH_ALL_INCLUDE_OFF_ONLY = 0x02;
+    private static final int SWITCH_ALL_INCLUDE_ON_OFF = 0xFF;
+    private boolean isGetSupported = true;
```

Comment:
This sort of thing largely needs to be done manually. The binding automatically configures things like associations, but  I don't think the binding can take a guess at what the user wants for the all on class - can it?

It's relatively easy to add this configuration so that it appears in HABmin if that's what you want, but that's another issue...